message ChatObjectInfo{
    required int32 id = 1;//角色id
    optional string nickname=2;//角色昵称
    optional int32 vip_level = 3;
    optional int32 guild_position = 4;
    optional int32 head = 5; //玩家头像
}

//登录 command:1001
message LoginToChatRequest{
	required ChatObjectInfo owner = 1;
}

// 发言
message ChatConectingRequest{
	required ChatObjectInfo owner = 1; // 发言角色信息
    required int32 channel = 2;//消息的频道  1.世界 2.公会 3.私聊 4.GM 5.系统通告 6.军团通告
    required string content = 3;//聊天内容
    optional ChatObjectInfo other = 4; // 私聊角色信息
    optional int32 guild_id = 5;//公会ID
    optional int32 vip_level = 6;
    optional int32 guild_position = 7;
}

//返回结果
message ChatResponse{
     required bool result = 1;
     optional int32 result_no = 2;
     optional int32 gag_time = 3;//禁言结束时间。－2 为永久禁言
}

//推送聊天信息
message chatMessageResponse{
    required int32 channel = 1;//消息的频道  1.世界 2.公会 3.私聊 4.GM 5.系统通告 6.军团通告
	required ChatObjectInfo owner = 2;
    required string content = 3;//消息内容
    optional int32 time = 4;
}

// 获取聊天记录 1004
message GetChatHistoryRequest{
    required int32 channel = 1;//消息的频道  1.世界 2.公会 3.私聊 4.GM 5.系统通告 6.军团通告
}

// 获取聊天记录 1004
message GetChatHistoryResponse{
    repeated chatMessageResponse world_chat_history=1;
    repeated chatMessageResponse guild_chat_history=2;
}

